nilearn/plotting/tests/test_surf_plotting.py

```diff
@@ -70,6 +94,91 @@ def test_set_view_plot_surf_matplotlib(hemi, view, expected_view_matplotlib):
            == expected_view_matplotlib)
 
 
+def test_surface_figure():
+    from nilearn.plotting.displays import SurfaceFigure
+    s = SurfaceFigure()
+    assert s.output_file is None
+    assert s.figure is None
+    with pytest.raises(NotImplementedError):
+        s.show()
+    with pytest.raises(ValueError, match="You must provide an output file"):
+        s._check_output_file()
+    s._check_output_file("foo.png")
+    assert s.output_file == "foo.png"
+    s = SurfaceFigure(output_file="bar.png")
+    assert s.output_file == "bar.png"
+
+
+@pytest.mark.skipif(PLOTLY_INSTALLED,
+                    reason='Plotly is installed.')
+def test_plotly_surface_figure_import_error():
+    """Test that an ImportError is raised when instantiating a
+    PlotlySurfaceFigure without having Plotly installed.
+    """
+    with pytest.raises(ImportError, match="Plotly is required"):
+        PlotlySurfaceFigure()
+
+
+@pytest.mark.skipif(not PLOTLY_INSTALLED or KALEIDO_INSTALLED,
+                    reason=("This test only runs if Plotly is "
+                            "installed, but not kaleido."))
+def test_plotly_surface_figure_savefig_error():
+    """Test that an ImportError is raised when saving a
+    PlotlySurfaceFigure without having kaleido installed.
+    """
+    with pytest.raises(ImportError, match="`kaleido` is required"):
+        PlotlySurfaceFigure().savefig()
+
+
+@pytest.mark.skipif(not PLOTLY_INSTALLED or not KALEIDO_INSTALLED,
+                    reason=("Plotly and/or kaleido not installed; "
+                            "required for this test."))
+def test_plotly_surface_figure():
+    ps = PlotlySurfaceFigure()
+    assert ps.output_file is None
+    assert ps.figure is None
+    ps.show()
+    with pytest.raises(ValueError, match="You must provide an output file"):
+        ps.savefig()
+    ps.savefig('foo.png')
+
+
+@pytest.mark.skipif(not PLOTLY_INSTALLED or not IPYTHON_INSTALLED,
+                    reason=("Plotly and/or Ipython is not installed; "
+                            "required for this test."))
+@pytest.mark.parametrize("renderer", ['png', 'jpeg', 'svg'])
+def test_plotly_show(renderer):
+    ps = PlotlySurfaceFigure(go.Figure())
+    assert ps.output_file is None
+    assert ps.figure is not None
+    with mock.patch("IPython.display.display") as mock_display:
+        ps.show(renderer=renderer)
+    assert len(mock_display.call_args.args) == 1
```

Comment:
these checks are slightly cryptic for non-experts. Briefly explain what is done here.